cmd/agent/app/install_service_windows.go

```diff
@@ -74,12 +74,14 @@ func exePath() (string, error) {
 	}
 	if filepath.Ext(p) == "" {
 		p += ".exe"
-		fi, err := os.Stat(p)
-		if err == nil {
+		fi, statErr := os.Stat(p)
+		if statErr == nil {
 			if !fi.Mode().IsDir() {
 				return p, nil
 			}
 			err = fmt.Errorf("%s is directory", p)
+		} else {
+			err = statErr
 		}
```

Comment:
This is a good catch, but I think the logic in this function is still a little muddled. I don't believe folders can have ".exe" endings, so checking if the file is a directory after that second call to `os.Stat` is unnecessary, and it means that the second `err = fmt.Errorf("%s is directory", p)` is unreachable... So maybe we could switch that second if block to something more along the lines of
```
if filepath.Ext(p) == "" {
   p += ".exe"
   if fi, err := os.Stat(p); err == nil {
      return p, nil
   }
}
```
to clean it up a little bit? Thoughts?

Reply:
Are you sure directories can never have `.exe` endings? I'm not familiar enough with Windows... @derekwbrown can you chime in here?

To be honest I only changed this because `ineffassign` (rightly) complained about the original `err` in `fi, err := os.Stat(p)` not being reachable at the very end of the function when we return an error.